### PR TITLE
logger: allow configuring what logs get sent to the remote logging service

### PR DESCRIPTION
**Pull request by bmwill**
Allow configuring what logs get sent to the remote logging service by respecting the `RUST_LOG_REMOTE` environment variable. If `RUST_LOG_REMOTE` is not set then fall back to look at the normal `RUST_LOG` environment variable.

### CODE REVIEW

**Review by sherry-x — DISMISSED** [review]

File: common/logger/src/diem_logger.rs [hunk 1]
```diff
@@ -30,6 +30,7 @@ use std::{
 };
 
 const RUST_LOG: &str = "RUST_LOG";
+const RUST_LOG_REMOTE: &str = "RUST_LOG_REMOTE";
```
Comment (aching) [inline common/logger/src/diem_logger.rs:33]: Should we add documentation about how users are supposed to config this?  Or document it elsewhere?

Comment (bmwill) [reply common/logger/src/diem_logger.rs:33]: I mean I can add a comment here, but AFAIK there's very little actual documentation around usage of our software. In that we don't really have the equivalent of a `man` page.

File: common/logger/src/diem_logger.rs [hunk 2]
```diff
@@ -239,7 +240,13 @@ impl DiemLoggerBuilder {
                 let mut filter_builder = Filter::builder();
 
                 if self.is_async && self.address.is_some() {
```
Comment (aching) [inline common/logger/src/diem_logger.rs:242]: The code looks good, but is there someway to test this?

File: common/logger/src/diem_logger.rs (hunk 1 above)
Comment (aching) [reply common/logger/src/diem_logger.rs:33]: I think a comment here about how the environment variables should be used would be good.

File: common/logger/src/diem_logger.rs (hunk 2 above)
Comment (aching) [reply common/logger/src/diem_logger.rs:242]: Also, is there a reason why the fallback to RUST_LOG if RUST_LOG_REMOTE isn't set?  Not sure why we need to do this.

Comment (bmwill) [reply common/logger/src/diem_logger.rs:242]: Tweaked a test to be able to ensure that the `RUST_LOG_REMOTE` env var is respected

File: common/logger/src/diem_logger.rs (hunk 1 above)
Comment (gregnazario) [reply common/logger/src/diem_logger.rs:33]: We can add to the documentation in the operator runbooks.  There is one for monitoring under full nodes.

**Review by gregnazario — APPROVED** [review]

File: common/logger/src/diem_logger.rs (hunk 2 above)
Comment (aching) [reply common/logger/src/diem_logger.rs:242]: > Also, is there a reason why the fallback to RUST_LOG if RUST_LOG_REMOTE isn't set? Not sure why we need to do this.

??

Comment (gregnazario) [reply common/logger/src/diem_logger.rs:242]: `RUST_LOG` is supposed to be the default variable for all logging.
`RUST_LOG_REMOTE` is the variable for remote logging.

We log two sources to local logs (e.g. pod logs) and to remote logs (e.g. vector / logstash / etc).

So, the fallback is if you set the variable for all the logs, you just get the same behavior everywhere

Comment (bmwill) [reply common/logger/src/diem_logger.rs:242]: Exactly, while in production we probably will always set both this is more about providing a sane default

Comment (aching) [reply common/logger/src/diem_logger.rs:242]: Got it, as long as that's well documented makes sense!

**Review by aching — APPROVED** [review]
Thanks for the quick fix @bmwill !

**Comment by sherry-x** [comment]
@bmwill let's cherry-pick this to the release branch and submit for gatekeeper review?

**Pull request merged**